### PR TITLE
Daniel Qu Firmware 102 Homework

### PR DESCRIPTION
**Pull request by danielq987**
Counter Project

### CODE REVIEW

File: projects/soft_timer/src/main.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  soft_timer_start_millis(COUNTER_B_DELAY, prv_timer_b_callback, counters, NULL);
+}
+
+int main() {
+  interrupt_init();
+  soft_timer_init();
+
+  Counters counters = { 0 };
+
+  soft_timer_start_millis(COUNTER_A_DELAY, prv_timer_a_callback, &counters, NULL);
```
Comment (AveryChiu64) [inline projects/soft_timer/src/main.c:37]: You should only use one soft timer and one callback function for both counters

File: projects/soft_timer/src/main.c [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+int main() {
+  interrupt_init();
+  soft_timer_init();
+
+  Counters counters = { 0 };
+
+  soft_timer_start_millis(COUNTER_A_DELAY, prv_timer_callback, &counters, NULL);
+
+  while (true) {
```
Comment (AveryChiu64) [inline projects/soft_timer/src/main.c:40]: We normally add a `wait();` in here so the processor can sleep and wait for an interrupt.

File: projects/soft_timer/src/main.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+// Values are in milliseconds.
+#define COUNTER_A_DELAY 500
```
Comment (AveryChiu64) [inline projects/soft_timer/src/main.c:9]: It would be good to add a suffix to this to indicate the units (ie COUNTER_A_DELAY_MS)

**Comment by mitchellostler** [comment]
Hey Daniel, for future reference, make sure that you get the CI build to pass for any PR that you submit. It's also good to resolve all the comments that have been put on your code, so that we know what you've addressed. Closing this now, as it's been open for ages

**Pull request closed without merging**